crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -17,6 +18,42 @@ use acvm::{
 use iter_extended::vecmap;
 use std::{borrow::Cow, collections::HashMap, hash::Hash};
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+/// High level Type descriptor for Variables.
+///
+/// Once can think of Expression/Witness/Const
+/// as low level types which can represent high level types.
+///
+/// An Expression can represent a u32 for example.
+/// We could store this information when we do a range constraint
+/// but this information is readily available by the caller so
+/// we allow the user to pass it in.
+pub(crate) struct AcirType(NumericType);
+
+impl AcirType {
+    /// Returns the bit size of the underlying type
+    fn bit_size(&self) -> u32 {
+        match self.0 {
+            NumericType::Signed { bit_size } => bit_size,
+            NumericType::Unsigned { bit_size } => bit_size,
+            NumericType::NativeField => FieldElement::max_num_bits(),
+        }
+    }
+
+    /// Returns a boolean type
+    fn boolean() -> Self {
+        AcirType(NumericType::Unsigned { bit_size: 1 })
+    }
+}
+impl From<SSAType> for AcirType {
```

Comment:
We _could_ change this from a From impl to a method in mod.rs. This is if we decide to make this module completely decoupled from the ssa code. We'd eventually need to change the inner type of AcirType too